diff -dur vlc-0.8.5/include/vlc_config.h vlc-0.8.5-democracy/include/vlc_config.h
--- vlc-0.8.5/include/vlc_config.h	2006-05-06 11:52:25.000000000 -0400
+++ vlc-0.8.5-democracy/include/vlc_config.h	2006-07-05 13:25:15.000000000 -0400
@@ -50,7 +50,7 @@
 #elif __APPLE__
 #  define CONFIG_DIR                    "Library/Preferences/VLC"
 #elif defined( WIN32 ) || defined( UNDER_CE )
-#  define CONFIG_DIR                    "vlc"
+#  define CONFIG_DIR                    "PCF-VLC"
 #else
 #  define CONFIG_DIR                    ".vlc"
 #endif
diff -dur vlc-0.8.5/modules/video_output/directx/events.c vlc-0.8.5-democracy/modules/video_output/directx/events.c
--- vlc-0.8.5/modules/video_output/directx/events.c	2006-05-06 11:52:17.000000000 -0400
+++ vlc-0.8.5-democracy/modules/video_output/directx/events.c	2006-07-05 13:25:59.000000000 -0400
@@ -88,6 +88,7 @@
     vlc_value_t val;
     int i_width, i_height, i_x, i_y;
     HMODULE hkernel32;
+    HWND parent;
 
     /* Initialisation */
     p_event->p_vout->pf_control = Control;
@@ -156,6 +157,11 @@
                 val.b_bool = VLC_TRUE;
                 var_Set( p_event->p_vout, "mouse-moved", val );
             }
+            parent = GetParent(msg.hwnd);
+            if(parent != NULL) {
+                PostMessage(parent, msg.message, msg.lParam, msg.wParam);
+            }
+            break;
 
         case WM_NCMOUSEMOVE:
             GetCursorPos( &mouse_pos );
@@ -204,7 +210,6 @@
             break;
 
         case WM_LBUTTONDBLCLK:
-            p_event->p_vout->p_sys->i_changes |= VOUT_FULLSCREEN_CHANGE;
             break;
 
         case WM_MBUTTONDOWN:
diff -dur vlc-0.8.5/modules/video_output/wingdi.c vlc-0.8.5-democracy/modules/video_output/wingdi.c
--- vlc-0.8.5/modules/video_output/wingdi.c	2006-05-06 11:52:17.000000000 -0400
+++ vlc-0.8.5-democracy/modules/video_output/wingdi.c	2006-07-05 13:25:51.000000000 -0400
@@ -1046,6 +1046,7 @@
                                 WPARAM wParam, LPARAM lParam )
 {
     vout_thread_t *p_vout;
+    HWND parent;
 
     if( message == WM_CREATE )
     {
@@ -1141,9 +1142,12 @@
         break;
 
     case WM_LBUTTONDOWN:
-        p_vout->p_sys->i_changes |= VOUT_FULLSCREEN_CHANGE;
         break;
     case WM_MOUSEMOVE:
+        parent = GetParent(hWnd);
+        if(parent != NULL) {
+            PostMessage(parent, message, wParam, lParam);
+        }
         break;
     case WM_LBUTTONUP:
         break;
diff -dur vlc-0.8.5/mozilla/vlcshell.cpp vlc-0.8.5-democracy/mozilla/vlcshell.cpp
--- vlc-0.8.5/mozilla/vlcshell.cpp	2006-05-06 11:52:25.000000000 -0400
+++ vlc-0.8.5-democracy/mozilla/vlcshell.cpp	2006-07-05 13:25:15.000000000 -0400
@@ -32,6 +32,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
+#include <direct.h> /* _getcwd */
 
 /* vlc stuff */
 #ifdef USE_LIBVLC
@@ -326,27 +327,25 @@
         };
 
 #elif defined(XP_WIN)
-        char *ppsz_argv[] = { NULL, "-vv" };
-        HKEY h_key;
-        DWORD i_type, i_data = MAX_PATH + 1;
+        char *ppsz_argv[] = { NULL, "-vv","--plugin-path", NULL, "--config", NULL };
         char p_data[MAX_PATH + 1];
-        if( RegOpenKeyEx( HKEY_LOCAL_MACHINE, "Software\\VideoLAN\\VLC",
-                          0, KEY_READ, &h_key ) == ERROR_SUCCESS )
-        {
-             if( RegQueryValueEx( h_key, "InstallDir", 0, &i_type,
-                                  (LPBYTE)p_data, &i_data ) == ERROR_SUCCESS )
-             {
-                 if( i_type == REG_SZ )
-                 {
-                     strcat( p_data, "\\vlc" );
-                     ppsz_argv[0] = p_data;
-                 }
-             }
-             RegCloseKey( h_key );
-        }
-
         if( !ppsz_argv[0] ) ppsz_argv[0] = "vlc";
 
+        /* Set plugin path relative to the current working directory */
+        if (_getcwd(p_data, MAX_PATH) == NULL) {
+          ppsz_argv[3] = "vlc-plugins";
+        } else {
+
+          strcat(p_data, "\\vlc");
+          ppsz_argv[0] = strdup(p_data);
+          _getcwd(p_data, MAX_PATH);
+          strcat(p_data, "\\vlc-plugins");
+          ppsz_argv[3] = strdup(p_data);
+          _getcwd(p_data, MAX_PATH);
+          strcat(p_data, "\\vlcrc");
+          ppsz_argv[5] = strdup(p_data);
+        }
+
 #else /* XP_MACOSX */
         char *ppsz_argv[] =
         {
@@ -373,7 +372,6 @@
 
         i_ret = VLC_Init( p_plugin->i_vlc, sizeof(ppsz_argv)/sizeof(char*),
                           ppsz_argv );
-
     }
 
     if( i_ret )
diff -dur vlc-0.8.5/src/misc/modules.c vlc-0.8.5-democracy/src/misc/modules.c
--- vlc-0.8.5/src/misc/modules.c	2006-05-06 11:52:13.000000000 -0400
+++ vlc-0.8.5-democracy/src/misc/modules.c	2006-07-05 13:25:15.000000000 -0400
@@ -2133,7 +2133,7 @@
 
     /* Code int size, pointer size and endianness in the filename */
     int32_t x = 0xbe00001e;
-    sprintf( psz_cachename, "plugins-%.2x%.2x%.2x.dat", sizeof(int),
+    sprintf( psz_cachename, "pcfplugins-%.2x%.2x%.2x.dat", sizeof(int),
              sizeof(void *), (unsigned int)((unsigned char *)&x)[0] );
     return psz_cachename;
 }
