--- platform/gtk-x11/platformcfg.py.orig	2006-06-03 19:34:58.000000000 +0200
+++ platform/gtk-x11/platformcfg.py	2006-06-03 19:34:06.000000000 +0200
@@ -71,7 +71,7 @@
     path = None
 
     if descriptor == prefs.MOVIES_DIRECTORY:
-        path = os.path.expanduser('~/Movies/Democracy')
+        path = os.path.expanduser('~/.democracy/Movies')
 
     elif descriptor == prefs.SUPPORT_DIRECTORY:
         path = os.path.expanduser('~/.democracy')
