Index: platform/gtk-x11/platformcfg.py
===================================================================
--- platform/gtk-x11/platformcfg.py     (revision 5511)
+++ platform/gtk-x11/platformcfg.py     (working copy)
@@ -92,14 +92,14 @@
     value = descriptor.default
 
     if descriptor == prefs.MOVIES_DIRECTORY:
-        value = os.path.expanduser('~/Movies/Miro')
+        value = os.path.expanduser('~/.miro/Movies')
         try:
             os.makedirs (value)
         except:
             pass
 
     elif descriptor == prefs.NON_VIDEO_DIRECTORY:
-        value = os.path.expanduser('~/Desktop')
+        value = os.path.expanduser('~/.miro/Nonvideo')
 
     elif descriptor == prefs.GETTEXT_PATHNAME:
         value = resources.path("../../locale")
Index: platform/gtk-x11/upgrade.py
===================================================================
--- platform/gtk-x11/upgrade.py (revision 5511)
+++ platform/gtk-x11/upgrade.py (working copy)
@@ -21,6 +21,27 @@
 import gconf
 
 def upgrade():
+    # gconf settings
+    client = gconf.client_get_default()
+    def _copy_gconf(src, dst):
+        for entry in client.all_entries(src):
+            entry_dst = dst + '/' + entry.key.split('/')[-1]
+            client.set(entry_dst, entry.value)
+        for subdir in client.all_dirs(src):
+            subdir_dst = dst + '/' + subdir.split('/')[-1]
+            _copy_gconf (subdir, subdir_dst)
+
+    if client.dir_exists ("/apps/democracy/player") and not client.dir_exists ("/apps/miro"):
+        _copy_gconf("/apps/democracy/player", "/apps/miro")
+        client.recursive_unset("/apps/democracy", 1)
+        if client.get("/apps/miro/MoviesDirectory") is None:
+            value = os.path.expanduser('~/.democracy/Movies')
+            client.set_string("/apps/miro/MoviesDirectory", value)
+            try:
+                os.makedirs (value)
+            except:
+                pass
+
     # dot directory
     src = os.path.expanduser('~/.democracy')
     dst = os.path.expanduser('~/.miro')
@@ -28,6 +49,26 @@
         shutil.move(src, dst)
         shutil.rmtree(os.path.join(dst, "icon-cache"), True)
 
+        # Restore movie directory to old location, since we moved the parent directory.
+        dir = client.get_string("/apps/miro/MoviesDirectory")
+        oldsupportdir = os.path.expanduser('~/.democracy')
+        newsupportdir = os.path.expanduser('~/.miro')
+        if dir is not None and dir.startswith(oldsupportdir):
+            olddir = dir
+            newdir = newsupportdir + dir[len(oldsupportdir):]
+            while olddir.endswith('/'):
+                olddir = olddir[:-1]
+            while newdir.endswith('/'):
+                newdir = newdir[:-1]
+            try:
+                os.makedirs(os.path.dirname(olddir))
+            except:
+                pass
+            try:
+                shutil.move(newdir, olddir)
+            except:
+                pass
+
     # autostart file
     config_home = os.environ.get ('XDG_CONFIG_HOME',
                                   '~/.config')
@@ -49,26 +90,3 @@
                 os.remove (old_file)
             except:
                 pass
-
-    # gconf settings
-
-    client = gconf.client_get_default()
-
-    def _copy_gconf(src, dst):
-        for entry in client.all_entries(src):
-            entry_dst = dst + '/' + entry.key.split('/')[-1]
-            client.set(entry_dst, entry.value)
-        for subdir in client.all_dirs(src):
-            subdir_dst = dst + '/' + subdir.split('/')[-1]
-            _copy_gconf (subdir, subdir_dst)
-
-    if client.dir_exists ("/apps/democracy/player") and not client.dir_exists ("/apps/miro"):
-        _copy_gconf("/apps/democracy/player", "/apps/miro")
-        client.recursive_unset("/apps/democracy", 1)
-        if client.get("/apps/miro/MoviesDirectory") is None:
-            value = os.path.expanduser('~/Movies/Democracy')
-            client.set_string("/apps/miro/MoviesDirectory", value)
-            try:
-                os.makedirs (value)
-            except:
-                pass
