Index: platform/gtk-x11/platformcfg.py
===================================================================
--- platform/gtk-x11/platformcfg.py	(revision 4934)
+++ platform/gtk-x11/platformcfg.py	(working copy)
@@ -92,14 +92,14 @@
     value = descriptor.default
 
     if descriptor == prefs.MOVIES_DIRECTORY:
-        value = os.path.expanduser('~/Movies/Miro')
+        value = os.path.expanduser('~/.miro/Movies')
         try:
             os.makedirs (value)
         except:
             pass
 
     elif descriptor == prefs.NON_VIDEO_DIRECTORY:
-        value = os.path.expanduser('~/Desktop')
+        value = os.path.expanduser('~/.miro/Nonvideo')
 
     elif descriptor == prefs.GETTEXT_PATHNAME:
         value = resources.path("../../locale")
Index: platform/gtk-x11/upgrade.py
===================================================================
--- platform/gtk-x11/upgrade.py	(revision 4934)
+++ platform/gtk-x11/upgrade.py	(working copy)
@@ -66,9 +66,30 @@
         _copy_gconf("/apps/democracy/player", "/apps/miro")
         client.recursive_unset("/apps/democracy", 1)
         if client.get("/apps/miro/MoviesDirectory") is None:
-            value = os.path.expanduser('~/Movies/Democracy')
+            value = os.path.expanduser('~/.democracy/Movies')
             client.set_string("/apps/miro/MoviesDirectory", value)
             try:
                 os.makedirs (value)
             except:
                 pass
+
+    # Restore movie directory to old location, since we moved the parent directory.
+    dir = client.get_string("/apps/miro/MoviesDirectory")
+    oldsupportdir = os.path.expanduser('~/.democracy')
+    newsupportdir = os.path.expanduser('~/.miro')
+    if dir.startswith(oldsupportdir):
+        olddir = dir
+        newdir = newsupportdir + dir[len(oldsupportdir):]
+        while olddir.endswith('/'):
+            olddir = olddir[:-1]
+        while newdir.endswith('/'):
+            newdir = newdir[:-1]
+        try:
+            os.makedirs(dirname(olddir))
+        except:
+            pass
+        try:
+            shutil.move(newdir, olddir)
+        except:
+            pass
+        
