--- portable/app.py.orig	2007-02-03 21:45:43.000000000 +0100
+++ portable/app.py	2007-02-03 21:46:07.000000000 +0100
@@ -522,7 +522,7 @@
             global delegate
             feed.setDelegate(delegate)
             feed.setSortFunc(sorts.item)
-            autoupdate.setDelegate(delegate)
+            # autoupdate.setDelegate(delegate)
             database.setDelegate(delegate)
             dialogs.setDelegate(delegate)
             
@@ -645,7 +645,7 @@
 
             util.print_mem_usage("Post-item reconnect memory check")
 
-            eventloop.addTimeout (30, autoupdate.checkForUpdates, "Check for updates")
+            # eventloop.addTimeout (30, autoupdate.checkForUpdates, "Check for updates")
             feed.expireItems()
 
             self.tabDisplay = TemplateDisplay('tablist','default',
