Index: portable/app.py
===================================================================
--- portable/app.py	(revision 5126)
+++ portable/app.py	(working copy)
@@ -45,7 +45,7 @@
 import storedatabase
 import subscription
 import downloader
-import autoupdate
+#import autoupdate
 import xhtmltools
 import guide
 import idlenotifier 
@@ -650,7 +650,7 @@
         global delegate
         feed.setDelegate(delegate)
         feed.setSortFunc(sorts.item)
-        autoupdate.setDelegate(delegate)
+        #autoupdate.setDelegate(delegate)
         database.setDelegate(delegate)
         dialogs.setDelegate(delegate)
         
@@ -808,7 +808,7 @@
 
         util.print_mem_usage("Post-item reconnect memory check")
 
-        eventloop.addTimeout (30, autoupdate.checkForUpdates, "Check for updates")
+        #eventloop.addTimeout (30, autoupdate.checkForUpdates, "Check for updates")
         feed.expireItems()
 
         self.tabDisplay = TemplateDisplay('tablist', 'default',
