Index: platform/gtk-x11/setup.py
===================================================================
--- platform/gtk-x11/setup.py	(revision 4924)
+++ platform/gtk-x11/setup.py	(working copy)
@@ -207,7 +207,7 @@
 if mozIncludeBase is None:
     raise ValueError("Can't find mozilla include base directory")
 for subdir in ['dom', 'gfx', 'widget', 'commandhandler', 'uriloader',
-            'webbrwsr']:
+            'webbrwsr', 'necko']:
     path = os.path.join(mozIncludeBase, subdir)
     mozilla_browser_options['include_dirs'].append(path)
 
