Index: platform/gtk-x11/plat/config.py
===================================================================
--- platform/gtk-x11/plat/config.py	(revision 6519)
+++ platform/gtk-x11/plat/config.py	(working copy)
@@ -103,13 +103,13 @@
     value = descriptor.default
 
     if descriptor == prefs.MOVIES_DIRECTORY:
-        value = os.path.expanduser('~/Movies/Miro')
+        value = os.path.expanduser('~/.miro/Movies')
         try:
             os.makedirs (value)
         except:
             pass
     elif descriptor == prefs.NON_VIDEO_DIRECTORY:
-        value = os.path.expanduser('~/Desktop')
+        value = os.path.expanduser('~/.miro/Nonvideo')
 
     elif descriptor == prefs.GETTEXT_PATHNAME:
         value = resources.path("../../locale")
Index: platform/gtk-x11/plat/upgrade.py
===================================================================
--- platform/gtk-x11/plat/upgrade.py	(revision 6519)
+++ platform/gtk-x11/plat/upgrade.py	(working copy)
@@ -32,6 +32,29 @@
 import gconf
 
 def upgrade():
+    # gconf settings
+
+    client = gconf.client_get_default()
+
+    def _copy_gconf(src, dst):
+        for entry in client.all_entries(src):
+            entry_dst = dst + '/' + entry.key.split('/')[-1]
+            client.set(entry_dst, entry.value)
+        for subdir in client.all_dirs(src):
+            subdir_dst = dst + '/' + subdir.split('/')[-1]
+            _copy_gconf (subdir, subdir_dst)
+
+    if client.dir_exists ("/apps/democracy/player") and not client.dir_exists ("/apps/miro"):
+        _copy_gconf("/apps/democracy/player", "/apps/miro")
+        client.recursive_unset("/apps/democracy", 1)
+        if client.get("/apps/miro/MoviesDirectory") is None:
+            value = os.path.expanduser('~/Movies/Democracy')
+            client.set_string("/apps/miro/MoviesDirectory", value)
+            try:
+                os.makedirs (value)
+            except:
+                pass
+
     # dot directory
     src = os.path.expanduser('~/.democracy')
     dst = os.path.expanduser('~/.miro')
@@ -61,25 +84,3 @@
             except:
                 pass
 
-    # gconf settings
-
-    client = gconf.client_get_default()
-
-    def _copy_gconf(src, dst):
-        for entry in client.all_entries(src):
-            entry_dst = dst + '/' + entry.key.split('/')[-1]
-            client.set(entry_dst, entry.value)
-        for subdir in client.all_dirs(src):
-            subdir_dst = dst + '/' + subdir.split('/')[-1]
-            _copy_gconf (subdir, subdir_dst)
-
-    if client.dir_exists ("/apps/democracy/player") and not client.dir_exists ("/apps/miro"):
-        _copy_gconf("/apps/democracy/player", "/apps/miro")
-        client.recursive_unset("/apps/democracy", 1)
-        if client.get("/apps/miro/MoviesDirectory") is None:
-            value = os.path.expanduser('~/Movies/Democracy')
-            client.set_string("/apps/miro/MoviesDirectory", value)
-            try:
-                os.makedirs (value)
-            except:
-                pass
