Index: portable/frontends/html/main.py
===================================================================
--- portable/frontends/html/main.py	(revision 6519)
+++ portable/frontends/html/main.py	(working copy)
@@ -62,7 +62,7 @@
     templates for the display (i.e. the Miro frontend for version 1.0)
     """
 
-    AUTOUPDATE_SUPPORTED = True
+    AUTOUPDATE_SUPPORTED = False
 
     def __init__(self):
         self.ignoreErrors = False
