Index: platform/gtk-x11/plat/screensaver.py
===================================================================
--- platform/gtk-x11/plat/screensaver.py (revision 9803)
+++ platform/gtk-x11/plat/screensaver.py (revision 9906)
@@ -77,8 +77,8 @@
 
     def should_use(self):
-        return call_xss('-time')
+        return self.call_xss('-time')
 
     def deactivate(self):
-        return call_xss('-deactivate')
+        return self.call_xss('-deactivate')
 
     def disable(self):
