--- ./portable/libtorrent/include/libtorrent/peer_id.hpp	2008-10-29 13:36:44.000000000 +0000
+++ ./portable/libtorrent/include/libtorrent/peer_id.hpp	2008-10-29 13:36:44.000000000 +0000
@@ -38,6 +38,7 @@
 #include <cctype>
 #include <algorithm>
 #include <string>
+#include <cstring>
 
 #include "libtorrent/config.hpp"
 #include "libtorrent/assert.hpp"
--- ./portable/libtorrent/src/file.cpp	2008-10-29 13:36:44.000000000 +0000
+++ ./portable/libtorrent/src/file.cpp	2008-10-29 13:36:44.000000000 +0000
@@ -57,6 +57,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <errno.h>
+#include <cstring>
 
 #include <boost/static_assert.hpp>
 // make sure the _FILE_OFFSET_BITS define worked
--- ./portable/libtorrent/src/entry.cpp	2008-10-29 13:36:44.000000000 +0000
+++ ./portable/libtorrent/src/entry.cpp	2008-10-29 13:36:44.000000000 +0000
@@ -34,6 +34,7 @@
 
 #include <algorithm>
 #include <iomanip>
+#include <ostream>
 #include "libtorrent/entry.hpp"
 #include "libtorrent/config.hpp"
