--- ./portable/flashscraper.py.orig	2008-10-29 13:36:44.000000000 +0000
+++ ./portable/flashscraper.py	2008-10-29 13:36:44.000000000 +0000
@@ -85,6 +85,5 @@
 
         url = u"http://www.youtube.com/get_video?video_id=%s&t=%s&eurl=&el=embedded&ps=default" % (videoID, token)
-        httpclient.grabURL(url, lambda x: callback(url, u"video/flv"),
-                           lambda x: _youTubeErrback(x, callback))
+        callback(url)
     except:
         logging.exception("youTubeCallbackStep2: unable to scrape YouTube Video URL")
